### ISSUE TITLE
Fine Grained State sample in documentation don't animate in React framework

### ISSUE THREAD

**Issue by jlopvet** [ag-grid/ag-grid#6180]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**

When calling columnApi.applyColumnState, the columns/grid don't animate nicely from one state to another in React whereas is does in pure JS, Angular or Vue.

React version : https://plnkr.co/edit/JKOjGBjsWZMHXOdM?open=index.jsx
JS version : https://plnkr.co/edit/1nNMCtBg9dj819fn?open=main.js&preview

**Expected behavior**

I would expect the same nice animation between states in React.

**Please tell us about your environment:**

Windows, running test in Plnkr and locally in node v16.15.1

* **AG Grid version:** 29.0.0

* **Browser:**

Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36
* **Language:** Running test in Plnkr and locally [TypeScript 4.8.2 | ES6]

**Comment by jlopvet** [comment]
The original sample can be found here in the official documentation : https://www.ag-grid.com/react-data-grid/column-state/#applying-partial-state
It has the same problem

**Comment by StephenCooper** [comment]
Thanks for reporting this. I have now raised this internally for further investigation.

**Comment by kiril-matev** [comment]
Hello @jlopvet,

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

Until this is fixed, you can work around this issue by switching to the legacy React renderer in AG Grid:
https://plnkr.co/edit/bqC2c2ZOokgVCE3Y?open=index.jsx

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-8081 When using React, applying new column state doesn't animate column headers**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!

**Comment by StephenCooper** [comment]
I can confirm that this has been resolved and will be part of our next release v29.2 scheduled for release before the end of the month.
